pages/docs/reference/whatsnew14.md

```diff
@@ -6,6 +6,293 @@ title: "What's New in Kotlin 1.4"
+    }
+    prop = "abc"
+    prop = "xyz"
+}
+```
+
+</div>
+
+#### SAM conversion for Java interfaces with different arguments
+
+Kotlin has supported SAM conversions for Java interfaces from the beginning, but there was one case that wasn’t supported, 
+which was sometimes annoying when working with existing Java libraries. If you called a Java method that took two SAM interfaces 
+as parameters, both arguments need to be either lambdas or regular objects. You couldn't pass one argument as a lambda and 
+another as an object. 
+
+The new algorithm fixes this issue, and you can pass a lambda instead of a SAM interface in any case, 
+which is the way you’d naturally expect it to work.
+
+<div class="sample" markdown="1" theme="idea" mode="java" data-highlight-only>
+
+```java
+// FILE: A.java
+public class A {
+    public void static foo(Runnable r1, Runnable r2) {}
+}
+```
+
+</div>
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+
+```kotlin
+// FILE: test.kt
+fun test(r1: Runnable) {
+    A.foo(r1) {}  // Works now
+}
+```
+
+</div>
+
+#### Java SAM interfaces in Kotlin
+
+Starting from Kotlin 1.4, you can use Java SAM interfaces in Kotlin. Previously, you could declare such a function `foo`
+only in Java code.
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+
+```kotlin
+fun foo(r1: java.lang.Runnable) {}
+
+fun test() { 
+    foo { } // OK
+}
+```
+
+</div>
+
+### Callable reference improvements
+
+Kotlin 1.4 supports more cases for using callable references:
+
+* References to functions with default argument values
+* Function references in `Unit`-returning functions 
+* References to functions with a variable number of arguments (`vararg`)
+* Suspend conversion on callable references in Kotlin/JVM
+
+#### References to functions with default argument values 
+
+Using callable references to functions with default argument values is now more convenient. For example, the callable reference 
+to the following `foo` function can be interpreted both as taking one `Int` argument or taking no arguments.
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+
+```kotlin
+fun foo(i: Int = 0): String = "$i!"
+
+fun apply1(func: () -> String): String = func()
+fun apply2(func: (Int) -> String): String = func(42)
+
+fun main() {
+    println(apply1(::foo))
+    println(apply2(::foo))
+}
+```
+
+</div>
+
+Previously, you should write additional overloads for the function `apply` to use the default argument values.
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+
+```kotlin
+// some new overload
+fun applyInt(func: (Int) -> String): String = func() 
+```
+
+</div>
+
+#### Function references in `Unit`-returning functions 
+
+Now you can use callable references to functions returning any type in `Unit`-returning functions. The Unit-returning 
+function `foo()` uses a callable reference to `returnsInt()` returning `Int` as a result.
```

Comment:
We can add that it worked for lambda arguments and now it works for callable references as well. For example:
```
fun foo(f: () -> Unit) { }

fun returnsInt(): Int = 42

fun main() {
    foo { returnsInt() } // it was OK before 1.4
    foo(::returnsInt) // since Kotlin 1.4, it works as well
}
```

Reply:
✅